### PR TITLE
#88562 Sequence diagram of the message flow with callback

### PR DESCRIPTION
**Pull request by tabman83**
[Preview here](http://www.plantuml.com/plantuml/png/ZPBBpjem48NtVeeHgrPgfVJIBbXKz8B4WaK5wWCCza0i7DlrXSlRTmmcYGgqFmkaPBupvynCPZ5XIBapOkEoLHZECFcEncnGxc5kbNVQfWcue2Z0vwQd-bejVIgbKz7wXDh2mhdzH7ZkhQNsoDpyG3Ojq2f3eOe-Da8wOqWcxMo45Q7gQnyQGLPnpy5oJU6W9NtBiJBl6Y56Bzz-kRMOma_D-RE7ZcB47S4sk0wIyrgAoZ0zQ5cnTAoe-8WPnsPgxV5eeRq-2z5QbmZSWSBVu5VvE-j0W3crGko2up214e_nwdqJt_1yyRM0iQhO3MkTmWyTFIRPzg5WgDu7AudP87hU99NLtKnZbf8RRBC1YLpYDYzD3V0qzrtoXv_DCCGG3VZtWFtVH00yqqFZmjNmU-kEXjIEG5NVY-B9nNzPCxeve2rs7Qz5a8a4sIeocfU46qFYNyOhyYuasA9cSwYgSv4CY-xX9IL8BS75cqvUX-kvPloNE_C7)

### CODE REVIEW

**Review by dawidkeshop — APPROVED** [review]
As for me it's very good!

File: docs/CS-CH-SD-webhook-callback.uml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+@startuml
+boundary Endpoint2 order 60
+boundary Endpoint1 order 50
+box "Captain Hook"
+participant EventHandler order 40
+collections Reader order 30
+participant Director order 20
```
Comment (chomick) [inline docs/CS-CH-SD-webhook-callback.uml:7]: We can probably ignore Director as he is not taking part in the flow

File: docs/CS-CH-SD-webhook-callback.uml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+group retry (20s, 30s)
+  EventHandler -> Endpoint1 : Dispatch message
+end
+  Endpoint1 -> EventHandler : Read response
+
+group successful case
+
+  group retry (20s, 30s)
+    EventHandler -> Endpoint2 : Dispatch response
```
Comment (chomick) [inline docs/CS-CH-SD-webhook-callback.uml:23]: What does endpoint 2 represent? Is it just to show we deliver to different endpoints?

File: docs/CS-CH-SD-webhook-callback.uml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+participant Director order 20
+endbox
+boundary ServiceBus order 10
+
+ServiceBus -> Reader : Pick up message from topic
+Reader -> ServiceBus : Lock message
+Reader -> EventHandler : Spawn handler
+
+group retry (20s, 30s)
+  EventHandler -> Endpoint1 : Dispatch message
```
Comment (chomick) [inline docs/CS-CH-SD-webhook-callback.uml:16]: Within the handler, do you think it makes sense to also show how we handle authentication before we send to the endpoint? Even on high level - we can cover in bit more detail in the description

File: docs/CS-CH-SD-webhook-callback.uml (hunk 2 above)
Comment (tabman83) [reply docs/CS-CH-SD-webhook-callback.uml:23]: Yes - I did not want to call it _callback endpoint_ to not generate confusion

Comment (chomick) [reply docs/CS-CH-SD-webhook-callback.uml:23]: Ah, it is a callback - can you name it so it is clear that the second call is for a callback, please?

**Review by chomick — APPROVED** [review]
Few small comments only

File: docs/CS-CH-SD-webhook-callback.uml (hunk 3 above)
Comment (tabman83) [reply docs/CS-CH-SD-webhook-callback.uml]: Added

**Pull request merged**